src/mercury-common/mercury/common/configuration.py

```diff
@@ -79,3 +84,368 @@ def get_configuration(filename):
+        Since argparse (by default) and os.environ can only be strings, we allow
+        options to specify a type
+        :param value: The value we are transforming
+        :param special_type: The 'type' we are transforming to
+        :type special_type: list, int, float, bool
+        :return: None
+        """
+
+        if value is None or special_type is None:
+            # Either value or special type are not set
+            # Check this here to prevent more branching up stream
+            return value
+
+        if type(special_type) != type:
+            raise MercuryConfigurationError(
+                f'{special_type} must be a type, not an object'
+            )
+        if special_type not in [list, int, float, bool]:
+            raise MercuryConfigurationError(
+                f'{special_type} is not support for conversion')
+
+        if isinstance(value, special_type):
+            # Type is already converted
+            return value
+
+        if not isinstance(value, str):
+            raise MercuryConfigurationError(
+                'Only string types are converted'
+            )
+
+        if special_type == list:
+            # Expected comma separated value, escaping is not supported
+            return value.split(',')
+
+        elif special_type == bool:
+            # if the string value is 0 or FALSE (any case) then the value is
+            # converted to False, otherwise True
+            return value.lower() not in ['0', 'false']
+        else:
+            try:
+                return special_type(value)
+            except ValueError:
+                raise MercuryConfigurationError(
+                    f'{value} cannot be converted to '
+                    f'{special_type}'
+                )
+
+    @staticmethod
+    def expand_option(namespace, option):
+        """ Expand an option, usually expressed in dotted notation, to a
+        namespace path. All but the last element, the value element, are
+        considered containers. For the option: nest1.nest2.nest3, nest1 and
+        nest2 would be initialized as new boxes, if they did not exist already.
+        nest3 would be initialized with the default value of the option
+        :param namespace: self.namespace reference
+        :param option: The option we are expanding
+        """
+        expanded_name = option['name'].split('.')
+        for container in expanded_name[:-1]:
+            if container not in namespace:
+                namespace[container] = Box()
+            # Check to see if the existing element is actually a container
+            elif not isinstance(namespace[container], dict):
+                raise MercuryConfigurationError(
+                    f'{container} has already been assigned a value, '
+                    f'{option["name"]} is invalid')
+            namespace = namespace[container]
+        namespace[expanded_name[-1]] = option['default']
+
+    def override_with_configuration(self, option):
```

Comment:
You can `DRY` these down to a single `_override` that there three call.

Reply:
👍